File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> num = new HashSet<>();
        for(int num1 : nums)
        {
            num.add(num1);
        }
        int longest = 0;
        for(int number : num)
        {
            if(!num.contains(number-1))
            {
                int length = 1;
                while(num.contains(number+length))
                {
                    length++;
                }
                longest = Math.max(longest,length);
            }
        }
        return longest;
    }
}

File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n=len(nums)
        dp=[-1]*(n+1)
        
        def rob1(i):
            if i>=n:return 0
            if dp[i]!=-1:return dp[i]

            current_robbery=nums[i]+rob1(i+2)
            skip_robbery=rob1(i+1)

            dp[i]=max(current_robbery,skip_robbery)
            return dp[i]
        return rob1(0)